packages/auth-password/index.js

```diff
@@ -95,8 +96,20 @@ class PasswordAuthStrategy {
 
   async _matchItem(item, args, secretFieldInstance) {
     const { secretField } = this.config;
+    let hash = item[secretField];
```

Comment:
```suggestion
    const { secretField } = this.config;
    const secret = args[secretField];
    if (item[secretField]) {
      const success = await secretFieldInstance.compare(secret, item[secretField]);
      return {
        success,
        message: success
          ? 'Authentication successful'
          : `[passwordAuth:secret:mismatch] The ${secretField} provided is incorrect`,
      };
    } else {
      hash = await secretFieldInstance.generateHash('password1234');
      await secretFieldInstance.compare(secret, hash);
      return {
        success: false,
        message:
          '[passwordAuth:secret:notSet] The item identified has no secret set so can not be authenticated',
      };
    }
```

Reply:
I have committed suggestions but had to fix them as well.